test/test_infmodel.py

```diff
@@ -166,8 +169,181 @@ def test_brute_pol_phase_marg(self):
         model.update(**self.q1)
         self.assertAlmostEqual(self.a2, model.loglr, delta=0.002)
 
+
+class TestWaveformErrors(unittest.TestCase):
```

Comment:
Thank you for adding a unittest! 

I think this mostly looks great. There was one minor comment. Otherwise, once you are happy with the PR and assuming all the tests now pass, I think this is good to go.